src/fields.jl

```diff
@@ -14,13 +14,21 @@ A field datatype for 1D objects.
 
 $(TYPEDFIELDS)
 """
-struct Field1D{LX<:AbstractLocation} <: AbstractField
+struct Field1D{LX<:AbstractLocation, G} <: AbstractField
     "Array with the values of the field."
-    data :: Array
+    data::AbstractArray
```

Comment:
@glwagner, should we better type the field data as `::OffsetArray` here? I'm wondering if `AbstractArray` will have some obvious performance drawback due to conversions between array types...?

Reply:
`OffsetArray` is not a concrete type either because the element type and number of dimensions isn't known.

You'd have to use something like `OffsetArray{Float64, Array{Float64, 1}}`.

But for `data` there's little advantage I think. Best to use a type parameter.

Reply:
Actually it would be

```julia
OffsetArray{Float64, 1, Array{Float64, 1}}
```

see here:

https://github.com/JuliaArrays/OffsetArrays.jl/blob/2a678a7223a4fc8f4c7925a7ade57432dbe1c514/src/OffsetArrays.jl#L112

Reply:
I admit I don't understand. Probably the word concrete in "concrete type" is technical, not just some metaphor of the cement, right?